code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -233,3 +233,147 @@
+
+	rage_power = min(1, rage_power) //Cap rage power so that we don't get way too insane.
+
+	var/rage_power_radius = CEILING(rage_power * 7, 1) //Define radius of the SFX
+
+	X.visible_message("<span class='danger'>\The [X] becomes frenzied, bellowing with a shuddering roar!</span>", \
+	"<span class='highdanger'>We bellow as our fury overtakes us! RIP AND TEAR!</span>")
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > RAVAGER_RAGE_SUPER_RAGE_THRESHOLD) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+	for(var/turf/affected_tiles AS in RANGE_TURFS(rage_power_radius, X.loc))
+		affected_tiles.Shake(4, 4, 1 SECONDS) //SFX
+
+	for(var/mob/living/L AS in cheap_get_living_near(X, rage_power_radius)) //Roar that applies cool SFX
+		if(L.stat == DEAD || !L.hud_used) //We don't care about the dead
+			continue
+
+		shake_camera(L, 1 SECONDS, 1)
+		L.Shake(4, 4, 1 SECONDS) //SFX
+
+		if(rage_power > RAVAGER_RAGE_SUPER_RAGE_THRESHOLD) //If we're super pissed it's time to get crazy
+
+			var/obj/screen/plane_master/floor/OT = L.hud_used.plane_masters["[FLOOR_PLANE]"]
+			var/obj/screen/plane_master/game_world/GW = L.hud_used.plane_masters["[GAME_PLANE]"]
+
+			addtimer(CALLBACK(OT, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+			GW.add_filter("rage_outcry", 2, radial_blur_filter(0.07))
+			animate(GW.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			OT.add_filter("rage_outcry", 2, radial_blur_filter(0.07))
+			animate(OT.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			addtimer(CALLBACK(GW, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+
+	X.add_filter("ravager_rage_outline", 5, outline_filter(1.5, COLOR_RED)) //Set our cool aura; also confirmation we have the buff
+
+	rage_plasma = min(X.xeno_caste.plasma_max - X.plasma_stored, X.xeno_caste.plasma_max * rage_power) //Calculate the plasma to restore (and take away later)
+	X.plasma_stored += rage_plasma //Regain a % of our maximum plasma scaling with rage
+
+	rage_sunder = min(X.sunder, rage_power * 100) //Set our temporary Sunder recovery
+	X.adjust_sunder(-1 * rage_sunder) //Restores up to 100 Sunder temporarily.
+
+	X.xeno_melee_damage_modifier += rage_power  //Set rage melee damage bonus
+
+	X.add_movespeed_modifier(MOVESPEED_ID_RAVAGER_RAGE, TRUE, 0, NONE, TRUE, X.xeno_caste.speed * 0.5 * rage_power) //Set rage speed bonus
+
+	//Too angry to be stunned/slowed/staggered/knocked down
+	ADD_TRAIT(X, TRAIT_STUNIMMUNE, RAGE_TRAIT)
```

Comment:
pretty sure ravager already cant be stunned

Reply:
there are ways to stun the ravager, but they're rare